### PR TITLE
Add stub of optional IPT support

### PR DESCRIPTION
**Pull request by icedevml**
drakvuf-bundle must be built from branch https://github.com/CERT-Polska/drakvuf/tree/sandbox-ipt-work and ran on a processor which supports IPT

Currently, the stub is able to run `ptxed` and report how many lines there were in disassembly.

### CODE REVIEW

File: drakcore/drakcore/postprocess/log_index.py [hunk 1]
```diff
@@ -11,8 +11,10 @@
 """
 import io
 import json
+import logging
```
Comment (chivay) [inline drakcore/drakcore/postprocess/log_index.py:14]: Unused import

File: drakrun/drakrun/main.py [hunk 2]
```diff
@@ -385,6 +394,9 @@ def process(self):
                                "-e", full_cmd,
                                "-c", cwd]
 
+                if self.config.config['drakrun'].get('enable_ipt', '0') == '1':
```
Comment (chivay) [inline drakrun/drakrun/main.py:397]: Maybe use `getboolean` : https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.getboolean

File: drakrun/drakrun/main.py [hunk 3]
```diff
@@ -433,6 +445,10 @@ def process(self):
                 self.log.exception("tcpdump doesn't exit cleanly after 60s")
 
         self.crop_dumps(os.path.join(outdir, 'dumps'), os.path.join(outdir, 'dumps.zip'))
+
+        if self.config.config['drakrun'].get('enable_ipt', '0') == '1':
```
Comment (chivay) [inline drakrun/drakrun/main.py:449]: ditto

File: drakrun/drakrun/scripts/cfg.template [hunk 4]
```diff
@@ -24,3 +24,4 @@ cpuid="host,htt=0"
 vga="stdvga"
 vif = [ 'type=ioemu,model=e1000,bridge=drak{{ VM_ID }}' ]
 disk = [ {{ DISKS }} ]
+processor_trace_buf_kb = 65536
```
Comment (chivay) [inline drakrun/drakrun/scripts/cfg.template:27]: Isn't it going to cause errors when running stock Xen? Are there checks for unknown config options?

File: drakcore/drakcore/postprocess/ipt_disasm.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+        injected_pid = inject_log["InjectedPid"]
+        resources["syscall.log"].download_to_file(str(tmpdir / "syscall.log"))
+
+        with open(str(tmpdir / "syscall.log"), "r") as f:
+            for line in f:
+                obj = json.loads(line)
+
+                if obj.get("PID") == injected_pid:
+                    injected_cr3 = int(obj["CR3"], 16)
```
Comment (chivay) [inline drakcore/drakcore/postprocess/ipt_disasm.py:156]:
```suggestion
                    injected_cr3 = hexint(obj["CR3"])
```

File: drakrun/drakrun/scripts/cfg.template (hunk 4 above)
Comment (icedevml) [reply drakrun/drakrun/scripts/cfg.template:27]: afaik this is no-op for stock Xen

Comment (chivay) [reply drakrun/drakrun/scripts/cfg.template:27]: :+1:

**Review by chivay — APPROVED** [review]

**Pull request merged**